File: msgraph-mail/typescript/src/users/mailFolders/messageRule.ts
import {SerializationWriter, ParseNode, Parsable} from '@microsoft/kiota-abstractions';
import {Entity} from '../entity';
import {MessageRuleActions} from './messageRuleActions';
import {MessageRulePredicates} from './messageRulePredicates';

export class MessageRule extends Entity implements Parsable<MessageRule> {
    public actions?: MessageRuleActions | undefined;
    public conditions?: MessageRulePredicates | undefined;
    public displayName?: string | undefined;
    public exceptions?: MessageRulePredicates | undefined;
    public hasError?: boolean | undefined;
    public isEnabled?: boolean | undefined;
    public isReadOnly?: boolean | undefined;
    public sequence?: number | undefined;
    public serialize (writer: SerializationWriter) : void {
        super.serialize(writer);
        writer.writeObjectValue<MessageRuleActions>("actions", this.actions);
        writer.writeObjectValue<MessageRulePredicates>("conditions", this.conditions);
        writer.writeStringValue("displayName", this.displayName);
        writer.writeObjectValue<MessageRulePredicates>("exceptions", this.exceptions);
        writer.writeBooleanValue("hasError", this.hasError);
        writer.writeBooleanValue("isEnabled", this.isEnabled);
        writer.writeBooleanValue("isReadOnly", this.isReadOnly);
        writer.writeNumberValue("sequence", this.sequence);
    };
    public deserializeFields () : Map<string, (item: MessageRule, node: ParseNode) => void> {
        return new Map<string, (item: MessageRule, node: ParseNode) => void>([...super.deserializeFields(),
            ["actions", (o, n) => { o.actions = n.getObjectValue<MessageRuleActions>(MessageRuleActions); }],
            ["conditions", (o, n) => { o.conditions = n.getObjectValue<MessageRulePredicates>(MessageRulePredicates); }],
            ["displayName", (o, n) => { o.displayName = n.getStringValue(); }],
            ["exceptions", (o, n) => { o.exceptions = n.getObjectValue<MessageRulePredicates>(MessageRulePredicates); }],
            ["hasError", (o, n) => { o.hasError = n.getBooleanValue(); }],
            ["isEnabled", (o, n) => { o.isEnabled = n.getBooleanValue(); }],
            ["isReadOnly", (o, n) => { o.isReadOnly = n.getBooleanValue(); }],
            ["sequence", (o, n) => { o.sequence = n.getNumberValue(); }],
        ]);
    };
}

File: msgraph-mail/dotnet/Users/MailFolders/MessageRule.cs
using Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.Linq;
namespace Graphdotnetv4.Users.MailFolders {
    public class MessageRule : Entity, IParsable<MessageRule> {
        public MessageRuleActions Actions { get; set; }
        public MessageRulePredicates Conditions { get; set; }
        public string DisplayName { get; set; }
        public MessageRulePredicates Exceptions { get; set; }
        public bool? HasError { get; set; }
        public bool? IsEnabled { get; set; }
        public bool? IsReadOnly { get; set; }
        public int? Sequence { get; set; }
        public new IDictionary<string, Action<MessageRule, IParseNode>> DeserializeFields => new Dictionary<string, Action<MessageRule, IParseNode>> {
            {
                "actions", (o,n) => { o.Actions = n.GetObjectValue<MessageRuleActions>(); }
            },
            {
                "conditions", (o,n) => { o.Conditions = n.GetObjectValue<MessageRulePredicates>(); }
            },
            {
                "displayName", (o,n) => { o.DisplayName = n.GetStringValue(); }
            },
            {
                "exceptions", (o,n) => { o.Exceptions = n.GetObjectValue<MessageRulePredicates>(); }
            },
            {
                "hasError", (o,n) => { o.HasError = n.GetBoolValue(); }
            },
            {
                "isEnabled", (o,n) => { o.IsEnabled = n.GetBoolValue(); }
            },
            {
                "isReadOnly", (o,n) => { o.IsReadOnly = n.GetBoolValue(); }
            },
            {
                "sequence", (o,n) => { o.Sequence = n.GetIntValue(); }
            },
        };
        public new void Serialize(ISerializationWriter writer) {
            base.Serialize(writer);
            writer.WriteObjectValue<MessageRuleActions>("actions", Actions);
            writer.WriteObjectValue<MessageRulePredicates>("conditions", Conditions);
            writer.WriteStringValue("displayName", DisplayName);
            writer.WriteObjectValue<MessageRulePredicates>("exceptions", Exceptions);
            writer.WriteBoolValue("hasError", HasError);
            writer.WriteBoolValue("isEnabled", IsEnabled);
            writer.WriteBoolValue("isReadOnly", IsReadOnly);
            writer.WriteIntValue("sequence", Sequence);
        }
    }
}
